packages/kbn-unified-data-table/src/components/source_document.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { Fragment } from 'react';
+import type {
+  DataTableRecord,
+  EsHitRecord,
+  FormattedHit,
+  ShouldShowFieldInTableHandler,
+} from '@kbn/discover-utils/src/types';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import type { FieldFormatsStart } from '@kbn/field-formats-plugin/public';
+import { formatHit } from '@kbn/discover-utils';
+import {
+  EuiDescriptionList,
+  EuiDescriptionListDescription,
+  EuiDescriptionListTitle,
+} from '@elastic/eui';
+import classnames from 'classnames';
+import { getInnerColumns } from '../utils/columns';
+
+const CELL_CLASS = 'unifiedDataTable__cellValue';
+
+export function SourceDocument({
```

Comment:
Could you please add a snapshot test? It's an important part of Discover and it would be great not to reduce the coverage by the refactoring.